dot/core/finality.go

```diff
@@ -56,24 +56,27 @@ func (s *Service) sendVoteMessages() {
 func (s *Service) sendFinalizationMessages() {
 	out := s.finalityGadget.GetFinalizedChannel()
 	for v := range out {
-		// TODO: safety
-		// update state.finalizedHead
+		log.Info("[core] finalized block!!!", "msg", v)
+		msg, err := v.ToConsensusMessage()
+		if err != nil {
+			log.Error("[core] failed to convert FinalizationMessage to ConsensusMessage", "msg", msg)
+			continue
+		}
+
+		// update finalized hash for this round in database
+		// TODO: this also happens in grandpa.finalize(); decide which is preferred
```

Comment:
my thought is it belongs in grandpa.finalize, no strong opinion just thinking it makes most sense there.

Reply:
I'll leave it as is for now, I'm leaning towards just doing it in core since that's where the block state updating should be done